src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useState } from 'react';
+import { Loader } from '../components/Loader';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { PeopleTable } from '../components/PeopleTable';
+import { useSearchParams } from 'react-router-dom';
+
+const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [searchParams] = useSearchParams();
+  const query = searchParams.get('query') || null;
+  const centuries = searchParams.getAll('centuries') || [];
+  const sex = searchParams.get('sex') || null;
+
+  useEffect(() => {
+    if (!isError) {
+      setIsLoading(true);
+      getPeople()
+        .then(result => {
+          const personWithParents = result.map(person => {
+            const modifiedPerson = { ...person };
+            const mam = result.find(p => p.name === person.motherName);
+            const dad = result.find(p => p.name === person.fatherName);
+
+            if (mam) {
+              modifiedPerson.mother = mam;
+            }
+
+            if (dad) {
+              modifiedPerson.father = dad;
+            }
+
+            return modifiedPerson;
+          });
+
+          setPeople(personWithParents);
+        })
+        .catch(() => {
+          setIsError(true);
+          setTimeout(() => {
+            setIsError(false);
+          }, 3000);
+        })
+        .finally(() => setIsLoading(false));
+    }
+  }, [isError]);
```

Comment:
The `useEffect` hook currently depends only on `isError`, which means it will not refetch data when other relevant parameters change. Consider adding dependencies such as `query`, `centuries`, and `sex` to ensure data is refetched when these parameters change.